pkg/util/provider/machinecontroller/secret.go

```diff
@@ -76,7 +76,7 @@ func (c *controller) reconcileClusterSecret(secret *corev1.Secret) error {
 			return err
 		}
 	} else {
-		if finalizers := sets.NewString(secret.Finalizers...); !finalizers.Has(DeleteFinalizerName) {
+		if finalizers := sets.NewString(secret.Finalizers...); !finalizers.Has(MCFinalizerName) {
```

Comment:
If I understand the solution, only `Secrets` get a machine-controller finalizer, and the rest of the objects gets machine-controller-manager finalizer. This could be little confusing. But I also understand the race condition, that both intree and OOT controllers are interested in setting/unsetting finalizers on the secret objects. 

wdys?

Overall, I suggest we consider this solution to be a stopgap, and once we get rid of the in-tree controllers[it doesn't need to worry about the provider-specific machine-classes], we align the finalizers to be the same.

Reply:
Yes agreed. Right now, it's a little tricky to migrate the existing finalizers on machine objects with the migration logic associated with it. But yes, in future let's stick to a consistent finalizer solution and this would be an intermediate solution until that.